### DISCUSSION TITLE
PnP "knockout-taxonomy"

### DISCUSSION THREAD

**Discussion by buszi2137** [microsoft-search/pnp-modern-search#3607] [General]
Hello, I wanted to build a sample called "knockout-taxonomy". When i try to install a required npm modules at node 7 this errors shows:
_npm ERR! Windows_NT 10.0.19045
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "i"
npm ERR! node v7.10.1
npm ERR! npm  v4.2.0

npm ERR! Unsupported URL Type: npm:@microsoft/teams-js@2.12.0
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\Lenovo\AppData\Roaming\npm-cache\_logs\2024-03-07T09_39_28_649Z-debug.log_
When i change to node 18 this error occurs:
npm ERR! code 1
npm ERR! path C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-sass
npm ERR! command failed
npm ERR! command C:\Windows\system32\cmd.exe /d /s /c node scripts/build.js
npm ERR! Building: C:\Program Files\nodejs\node.exe C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-gyp\bin\node-gyp.js rebuild --verbose --libsass_ext= --libsass_cflags= --libsass_ldflags= --libsass_library=
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp verb cli [
npm ERR! gyp verb cli   'C:\\Program Files\\nodejs\\node.exe',
npm ERR! gyp verb cli   'C:\\Users\\Lenovo\\Desktop\\knockout-taxonomy\\node_modules\\node-gyp\\bin\\node-gyp.js',
npm ERR! gyp verb cli   'rebuild',
npm ERR! gyp verb cli   '--verbose',
npm ERR! gyp verb cli   '--libsass_ext=',
npm ERR! gyp verb cli   '--libsass_cflags=',
npm ERR! gyp verb cli   '--libsass_ldflags=',
npm ERR! gyp verb cli   '--libsass_library='
npm ERR! gyp verb cli ]
npm ERR! gyp info using node-gyp@3.8.0
npm ERR! gyp info using node@18.19.1 | win32 | x64
npm ERR! gyp verb command rebuild []
npm ERR! gyp verb command clean []
npm ERR! gyp verb clean removing "build" directory
npm ERR! gyp verb command configure []
npm ERR! gyp verb check python checking for Python executable "C:\Users\Lenovo\.windows-build-tools\python27\python.exe" in the PATH
npm ERR! gyp verb `which` succeeded C:\Users\Lenovo\.windows-build-tools\python27\python.exe C:\Users\Lenovo\.windows-build-tools\python27\python.exe
npm ERR! gyp verb check python version `C:\Users\Lenovo\.windows-build-tools\python27\python.exe -c "import sys; print "2.7.15
npm ERR! gyp verb check python version .%s.%s" % sys.version_info[:3];"` returned: %j
npm ERR! gyp verb get node dir no --target version specified, falling back to host node version: 18.19.1
npm ERR! gyp verb command install [ '18.19.1' ]
npm ERR! gyp verb install input version string "18.19.1"
npm ERR! gyp verb install installing version: 18.19.1
npm ERR! gyp verb install --ensure was passed, so won't reinstall if already installed
npm ERR! gyp verb install version is already installed, need to check "installVersion"
npm ERR! gyp verb got "installVersion" 9
npm ERR! gyp verb needs "installVersion" 9
npm ERR! gyp verb install version is good
npm ERR! gyp verb get node dir target node version installed: 18.19.1
npm ERR! gyp verb build dir attempting to create "build" dir: C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-sass\build
npm ERR! gyp verb build dir "build" dir needed to be created? C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-sass\build
npm ERR! gyp verb find vs2017 Found installation at: C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools
npm ERR! gyp verb find vs2017   - Found Microsoft.VisualStudio.Component.Windows10SDK.17763
npm ERR! gyp verb find vs2017   - Found Microsoft.VisualStudio.Component.VC.Tools.x86.x64
npm ERR! gyp verb find vs2017   - Found Microsoft.VisualStudio.VC.MSBuild.Base
npm ERR! gyp verb find vs2017   - Using this installation with Windows 10 SDK
npm ERR! gyp verb find vs2017 using installation: C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools
npm ERR! gyp verb build/config.gypi creating config file
npm ERR! gyp verb build/config.gypi writing out config file: C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-sass\build\config.gypi
npm ERR! (node:23572) [DEP0150] DeprecationWarning: Setting process.config is deprecated. In the future the property will be read-only.
npm ERR! (Use `node --trace-deprecation ...` to show where the warning was created)
npm ERR! gyp verb config.gypi checking for gypi file: C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-sass\config.gypi
npm ERR! gyp verb common.gypi checking for gypi file: C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-sass\common.gypi
npm ERR! gyp verb gyp gyp format was not specified; forcing "msvs"
npm ERR! gyp info spawn C:\Users\Lenovo\.windows-build-tools\python27\python.exe
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   'C:\\Users\\Lenovo\\Desktop\\knockout-taxonomy\\node_modules\\node-gyp\\gyp\\gyp_main.py',
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'msvs',
npm ERR! gyp info spawn args   '-G',
npm ERR! gyp info spawn args   'msvs_version=2015',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   'C:\\Users\\Lenovo\\Desktop\\knockout-taxonomy\\node_modules\\node-sass\\build\\config.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   'C:\\Users\\Lenovo\\Desktop\\knockout-taxonomy\\node_modules\\node-gyp\\addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   'C:\\Users\\Lenovo\\.node-gyp\\18.19.1\\include\\node\\common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=C:\\Users\\Lenovo\\.node-gyp\\18.19.1',
npm ERR! gyp info spawn args   '-Dnode_gyp_dir=C:\\Users\\Lenovo\\Desktop\\knockout-taxonomy\\node_modules\\node-gyp',
npm ERR! gyp info spawn args   '-Dnode_lib_file=C:\\Users\\Lenovo\\.node-gyp\\18.19.1\\<(target_arch)\\node.lib',
npm ERR! gyp info spawn args   '-Dmodule_root_dir=C:\\Users\\Lenovo\\Desktop\\knockout-taxonomy\\node_modules\\node-sass',
npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! gyp info spawn args   '--depth=.',
npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! gyp info spawn args   '--generator-output',
npm ERR! gyp info spawn args   'C:\\Users\\Lenovo\\Desktop\\knockout-taxonomy\\node_modules\\node-sass\\build',
npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! Traceback (most recent call last):
npm ERR!   File "C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-gyp\gyp\gyp_main.py", line 16, in <module>
npm ERR!     sys.exit(gyp.script_main())
npm ERR!   File "C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-gyp\gyp\pylib\gyp\__init__.py", line 545, in script_main
npm ERR!     return main(sys.argv[1:])
npm ERR!   File "C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-gyp\gyp\pylib\gyp\__init__.py", line 538, in main
npm ERR!     return gyp_main(args)
npm ERR!   File "C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-gyp\gyp\pylib\gyp\__init__.py", line 514, in gyp_main
npm ERR!     options.duplicate_basename_check)
npm ERR!   File "C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-gyp\gyp\pylib\gyp\__init__.py", line 130, in Load
npm ERR!     params['parallel'], params['root_targets'])
npm ERR!   File "C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-gyp\gyp\pylib\gyp\input.py", line 2783, in Load
npm ERR!     variables, includes, depth, check, True)
npm ERR!   File "C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-gyp\gyp\pylib\gyp\input.py", line 399, in LoadTargetBuildFile
npm ERR!     includes, True, check)
npm ERR!   File "C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-gyp\gyp\pylib\gyp\input.py", line 271, in LoadOneBuildFile
npm ERR!     aux_data, includes, check)
npm ERR!   File "C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-gyp\gyp\pylib\gyp\input.py", line 308, in LoadBuildFileIncludesIntoDict
npm ERR!     LoadOneBuildFile(include, data, aux_data, None, False, check),
npm ERR!   File "C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-gyp\gyp\pylib\gyp\input.py", line 251, in LoadOneBuildFile
npm ERR!     None)
npm ERR!   File "C:\Users\Lenovo\.node-gyp\18.19.1\include\node\common.gypi", line 1
npm ERR!     e_data_file_flag%': 1
npm ERR!                         ^
npm ERR! SyntaxError: EOL while scanning string literal
npm ERR! gyp ERR! configure error
npm ERR! gyp ERR! stack Error: `gyp` failed with exit code: 1
npm ERR! gyp ERR! stack     at ChildProcess.onCpExit (C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-gyp\lib\configure.js:345:16)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:517:28)
npm ERR! gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:292:12)
npm ERR! gyp ERR! System Windows_NT 10.0.19045
npm ERR! gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\Lenovo\\Desktop\\knockout-taxonomy\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
npm ERR! gyp ERR! cwd C:\Users\Lenovo\Desktop\knockout-taxonomy\node_modules\node-sass
npm ERR! gyp ERR! node -v v18.19.1
npm ERR! gyp ERR! node-gyp -v v3.8.0
npm ERR! gyp ERR! not ok
npm ERR! Build failed with error code: 1
Do you know what should i do to get rid of these errors?

**Comment by wobba** [comment]
I don't see the relevance to this project with your error.

**Reply by buszi2137** [reply]
Do you know how to fix them? What should I install?
